exercises/custom-set/.meta/src/reference/kotlin/CustomSet.kt

```diff
@@ -0,0 +1,25 @@
+class CustomSet(private var set: Set<Int> = setOf()) {
+
+    constructor(vararg elements: Int) : this(elements.toSet())
+
+    fun isEmpty(): Boolean = set.isEmpty()
+
+    fun isSubset(other: CustomSet): Boolean = other.set.containsAll(set)
+
+    fun isDisjoint(other: CustomSet): Boolean = set.intersect(other.set).isEmpty()
+
+    fun contains(other: Int): Boolean = set.contains(other)
+
+    fun intersection(other: CustomSet): CustomSet = CustomSet(set.intersect(other.set))
+
+    fun add(other: Int) {
+        set = set + other
```

Comment:
Here is a tradeoff: 
  - define `set` as a `var` and reassign it here fully, 
  - or let it be *final* `val` inited as  `mutableSetOf()` and use `set.add()` in this method.

The second approach is always prefferable, as the class becomes *less mutable* - still not thread safe but at least represents always the sampe single entity in memory (set in our case).

Reply:
I played with both but couldn't decide :) I will change it to mutableSet and strip the constructor from the empty class as proposed!